tensorflow/compiler/tf2tensorrt/convert/utils.h

```diff
@@ -190,6 +191,27 @@ Status TrtDimsToTensorShape(const nvinfer1::Dims trt_dims,
   return Status::OK();
 }
 
+template <typename T>
+StatusOr<T> StaticDimsVolumeSafe(absl::Span<const int32_t> span) {
+  TRT_ENSURE(HasStaticShape(span));
+  int64_t prod = absl::c_accumulate(span, int64_t(1), std::multiplies<>());
+  TRT_ENSURE(prod <= std::numeric_limits<T>::max() && prod >= 0);
+  return static_cast<T>(prod);
+}
+
+template <typename T>
+T TrtDimsVolumeUnsafe(const nvinfer1::Dims& dims, int32_t offset = 0,
```

Comment:
I think it was more because it couldn't handle some "scalar shape" edge case. I'm going to rebase this on the PR for adding `DimsAdapter` since that one will merge before this and these functions go away.